src/corell/src/pool.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 The Gfx-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ops::{DerefMut};
+use {CommandPool, CommandQueue};
+pub use queue::{GeneralQueue, GraphicsQueue, ComputeQueue, TransferQueue};
+
+/// General command pool can allocate general command buffers.
+pub trait GeneralCommandPool: CommandPool {
+    fn from_queue<Q>(queue: &mut Q, capacity: usize) -> Self
+        where Q: Into<GeneralQueue<<Self as CommandPool>::Queue>> +
```

Comment:
Under no circumstances, but makes the implementation a bit easier for the backends.
Can change it if you prefer.